net/devif/devif_callback.c

```diff
@@ -165,6 +143,48 @@ static void devif_callback_free(FAR struct net_driver_s *dev,
             }
         }
 
+      /* check if the callback structure has DEVIF_CB_DONT_FREE,it indicates
+       * the callback can't be free immediately,setting DEVIF_CB_PEND_FREE
+       * flag with the callback,it indicates the callback will be free
+       * finally
+       */
+
+      if (cb->free_flags & DEVIF_CB_DONT_FREE)
+        {
+          cb->free_flags |= DEVIF_CB_PEND_FREE;
```

Comment:
you should remove the cb from device list and keep the cb do not add to freelist or free